src/main/java/io/odpf/firehose/sink/log/LogSinkFactory.java

```diff
@@ -29,7 +31,14 @@
      */
     public static Sink create(Map<String, String> configuration, StatsDReporter statsDReporter, StencilClient stencilClient) {
         AppConfig appConfig = ConfigFactory.create(AppConfig.class, configuration);
-        KeyOrMessageParser parser = new KeyOrMessageParser(stencilClient.getParser(appConfig.getInputSchemaProtoClass()), appConfig);
-        return new LogSink(parser, new Instrumentation(statsDReporter, LogSink.class));
+        InputSchemaDataType inputSchemaDataTye = appConfig.getInputSchemaDataTye();
+        switch (inputSchemaDataTye) {
```

Comment:
Can we create an abstraction out of sinks? basically, a sink should not worry about the input data type

Reply:
how about [here](https://github.com/odpf/firehose/blob/main/src/main/java/io/odpf/firehose/consumer/kafka/FirehoseKafkaConsumer.java#L60)?

basically instead of different parsers, how about different Message itself? 

Message can be common interface, and its implementations can have different methods for parsing